docs/api/io/UnbufferedSerial.md

```diff
@@ -0,0 +1,24 @@
+# UnbufferedSerial
+
+<span class="images">![](https://os.mbed.com/docs/development/mbed-os-api-doxy/classmbed_1_1_unbuffered_serial.png)<span>`UnbufferedSerial` class hierarchy</span></span>
+
+The `UnbufferedSerial` class provides UART functionality with an API similar to the [`BufferedSerial`](./BufferedSerial.md) class. The classes also share the same default configurations.
+
+Unlike the `BufferedSerial` class, the `UnbufferedSerial` class does not use intermediary buffers to store bytes to transmit to or read from the hardware. The user application is responsible for processing each byte as they are received. The method to read data returns only one byte for every call. It is therefore suitable when more control is required and for use in interrupt handlers with the RTOS. The class can however be used to write multiple bytes at once as it locks the API usage in the RTOS to ensure all bytes are written to the hardware.
```

Comment:
Perhaps also add that it has no locks so it is safe to use in interrupts provided that it is the only instance that is using that serial port.

Reply:
https://github.com/ARMmbed/mbed-os-5-docs/compare/934953559ca74a40ae8258e144792d1f2efc79af..32c308336088d2e5634585b28f720375c550dad7